aiosmtpd/smtp.py

```diff
@@ -380,7 +380,11 @@ async def _handle_client(self):
                 # verify that it was actually self._reader that was reset.
                 log.info('Connection lost during _handle_client()')
                 self._writer.close()
-                raise
+                break
```

Comment:
I'm -1 on replacing `raise` with `break`

[Documentation since 3.7](https://docs.python.org/3.7/library/asyncio-exceptions.html#asyncio.CancelledError) clearly states that:

> In almost all situations the exception must be re-raised.

I'm not sure if our situation suits the small sliver of situation where we shouldn't re-raise.

In fact, testing the test cases on my computer, all tests still pass even if I revert `break` back to `raise`

Reply:
Here's where `raise` causes a problem.

asyncio.CancelledError was a subclass of Exception before Python 3.8.
So if use `raise`, error will be captured again.
But on the second capture it calls `self.push()` to write to the closed socket.
This is why issue #127 happened.

Reply:
What if we put in a flag there?

Something like this:

```python
# outside try block
was_cancelled_error = False
...
    except asyncio.CancelledError:
        ...
        was_cancelled_error = True
        raise
    ...
    except Exception as error:
        if was_cancelled_error:
            raise
        ...
```

Therefore we still follow the asyncio recommendation of re-raising `CancelledError`, but prevent the broad except from firing its contents.

Reply:
That said, out of curiosity I ran this short snippet:

```python
class CustomException(Exception):
    pass


try:
    raise CustomException
except CustomException:
    raise
except Exception:
    print("recaught the exception")
```

And the `raise` within the first except block was never caught by the second block.

Tested on Ubuntu with Python 3.6, 3.7, and 3.8.

Reply:
I've done more testing and apparently you're right.

issue #127 should be the same as #162.
The error caused by calling `self.push()` with `handle_exception()` after the client disconnected.

Here I'll revert to `raise` and change the description.